server/service/integration_enterprise_test.go

```diff
@@ -10131,6 +10131,177 @@ func (s *integrationEnterpriseTestSuite) TestHostSoftwareInstallResult() {
+
+	// get the anonymous script result details, still works
+	scriptRes = getScriptResultResponse{}
+	s.DoJSON("GET", "/api/latest/fleet/scripts/results/"+scriptExecID, nil, http.StatusOK, &scriptRes)
+	require.Equal(t, scriptExecID, scriptRes.ExecutionID)
+	require.Equal(t, host.ID, scriptRes.HostID)
+	require.Equal(t, "ok", scriptRes.Output)
+	require.NotNil(t, scriptRes.ExitCode)
+	require.EqualValues(t, 0, *scriptRes.ExitCode)
+
+	// get the saved script result details, still works
+	scriptRes = getScriptResultResponse{}
+	s.DoJSON("GET", "/api/latest/fleet/scripts/results/"+savedScriptExecID, nil, http.StatusOK, &scriptRes)
+	require.Equal(t, savedScriptExecID, scriptRes.ExecutionID)
+	require.Equal(t, host.ID, scriptRes.HostID)
+	require.Equal(t, "saved", scriptRes.Output)
+	require.NotNil(t, scriptRes.ExitCode)
+	require.EqualValues(t, 0, *scriptRes.ExitCode)
+
+	// get the software install details, still works
+	softwareRes = getSoftwareInstallResultsResponse{}
+	s.DoJSON("GET", "/api/latest/fleet/software/install/results/"+installID, nil, http.StatusOK, &softwareRes)
+	require.Equal(t, installID, softwareRes.Results.InstallUUID)
+	require.Equal(t, host.ID, softwareRes.Results.HostID)
+	require.Equal(t, payload.Filename, softwareRes.Results.SoftwarePackage)
+	require.NotNil(t, softwareRes.Results.Output)
+	require.Equal(t, "Installing software...\nSuccess\nok", *softwareRes.Results.Output)
+	require.NotNil(t, softwareRes.Results.PreInstallQueryOutput)
+	require.EqualValues(t, "Query returned result\nProceeding to install...", *softwareRes.Results.PreInstallQueryOutput)
+	// we don't render the exit codes in the JSON response, so they are null
+	require.Nil(t, softwareRes.Results.InstallScriptExitCode)
+
+	// delete the software installer/named script
+	s.Do("DELETE", fmt.Sprintf("/api/latest/fleet/software/%d/package", titleID), nil, http.StatusNoContent, "team_id", fmt.Sprint(tm.ID))
+	s.Do("DELETE", fmt.Sprintf("/api/latest/fleet/scripts/%d", savedScriptID), nil, http.StatusNoContent)
+
+	// get the saved script result details, still works because the saved script
+	// is a "soft-reference", when deleted the results become essentially results
+	// for an anonymous script (i.e. the script_id FK is "ON DELETE SET NULL").
+	scriptRes = getScriptResultResponse{}
+	s.DoJSON("GET", "/api/latest/fleet/scripts/results/"+savedScriptExecID, nil, http.StatusOK, &scriptRes)
+	require.Equal(t, savedScriptExecID, scriptRes.ExecutionID)
+	require.Equal(t, host.ID, scriptRes.HostID)
+	require.Equal(t, "saved", scriptRes.Output)
+	require.NotNil(t, scriptRes.ExitCode)
+	require.EqualValues(t, 0, *scriptRes.ExitCode)
+
+	// get the software install details, does not work because the installer does
+	// not exist anymore.
+	softwareRes = getSoftwareInstallResultsResponse{}
+	s.DoJSON("GET", "/api/latest/fleet/software/install/results/"+installID, nil, http.StatusNotFound, &softwareRes)
```

Comment:
Note that deleting a software installer will still result in not being able to view the activity details. This PR only handles the host's soft-deletion.